### PR TITLE
Shell Verifier rewrite

### PR DESCRIPTION
**Pull request by thewyzard44**
Signed-off-by: Sean Zachariasen <thewyzard@hotmail.com>

Addressing #1209

This one needs some love.  This initial commit makes the options merge correctly, but the rest of the verifier needs touched up in some surrounding areas.  There's incongruity between 2 different methods of local execution and no effort has been applied by the OA to make sure the environment gets applied when running remotely

### CODE REVIEW

**Comment by coderanger** [comment]
What is the end goal here?

**Comment by thewyzard44** [comment]
I want to remove this `run_command` method https://github.com/test-kitchen/test-kitchen/blob/master/lib/kitchen/verifier/shell.rb#L55-L62.  It is not called in any normal command flow.

If anyone is calling that manually (and I'd be surprised if you are because it is broken) let me know.

I will try to deprecate it instead, if the new deprecation framework will allow this.

File: lib/kitchen/verifier/shell.rb [hunk 1]
```diff
@@ -73,8 +102,29 @@ def sleep_if_set
         end
       end
 
+      # Merges environment variables with :shellout_opts as specified in the config
+      #
+      # @return [Hash] options to be passed to shellout
+      # @api private
+      def shellout_opts
+        config[:shellout_opts].dup.tap do |options|
```
Comment (thewyzard44) [inline lib/kitchen/verifier/shell.rb]: cover 'if config[:shellout_opts].nil?`

**Comment by thewyzard44** [comment]
apologies - I answered coderanger OOB...  but in short, I'd like this verifier to work locally, or remotely, and to pass the expected environment when and wherever it runs.

It may be prudent to use a `script:` value in parity with the shell provisioner, if you just want to run a shell script.  But RFC, it'll have issues similar to #1381.

And so should this rewrite include a `script:` option?  Or should it be left as-is and expect the end user to spectify `sh -c myscript` as an option to `command:`

This is a forward moving question for the kitchen team in that they didn't even really like `chmod +x myscript` but took it due to lack of options....   to quote coderanger "sigh windows"

And also a community question for those vested, or interested, in a good shell verifier

**Comment by cheeseplus** [comment]
The community we have is the community that shows up with opinions and PRs so I think so long as we're moving it forward in a "less bad" direction then we are good.

**Comment by damacus** [comment]
Hey @thewyzard44 do we still want this PR and how can we help you get this finished?

**Pull request closed without merging**